test/pipelinerun_test.go

```diff
@@ -33,12 +34,102 @@ import (
 	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
 )
 
+// dumpAndTearDown will, when a test fails, print out all CRD objects as YAML
+// in the relevant namespace before tearing the namespace down. It will also
+// get the logs of all pods in the namespace. If the tests do not fail then the
+// namespace is simply torn down.
+func dumpAndTearDown(t *testing.T, logger *logging.BaseLogger, cs *clients, namespace string) {
+	defer tearDown(logger, cs.KubeClient, namespace)
+	if !t.Failed() {
+		return
+	}
+
+	var output []byte
+
+	printOrAdd := func(kind, name string, i interface{}) {
+		bs, err := yaml.Marshal(i)
+		if err != nil {
+			logger.Fatalf("couldn't marshal %s to yaml: %v", kind, err)
+		} else {
+			output = append(output, []byte("\n---\n")...)
+			output = append(output, bs...)
+		}
+	}
+	pos, err := cs.KubeClient.Kube.CoreV1().Pods(namespace).List(metav1.ListOptions{})
```

Comment:
it might help to start out the debug output with a header so we know what we're looking at (when i tried to see this in the logs I initially got confused and started looking at the info dumped by the e2e scripts which starts with the header:

```
I1024 13:43:43.408] ***************************************
I1024 13:43:43.409] ***           TEST FAILED           ***
I1024 13:43:43.409] ***    Start of information dump    ***
I1024 13:43:43.409] ***************************************
```